### PR TITLE
PROD-81: config i18n

### PR DESCRIPTION
**Pull request by shevchukm**
# Reviewers
- [x] @vikabelous
- [x] @okohuch
- [x] @anastasiabalaban 
- [x] @sashamuliar
# Jira task
https://ssu-jira.softserveinc.com/browse/PROD-81
# Description
config i18n, add translations files, add gem 'rails-i18n', add method to application_controller, add configs to application
# Checklist:
- [x] Rubocop doesn't raise any warnings and offenses
- [x] All specs pass
- [x] New features are covered with new specs
- [x] PR is approved by 2 team mates
- [x] PR is approved by 2 experts

### CODE REVIEW

**Comment by okohuch** [comment]
@shevchukm How user can change the locale?

**Comment by shevchukm** [comment]
@okohuch by pass <locale=uk> in params....  I have not any views for config them....

**Review by sashamuliar — APPROVED** [review]

File: config/locales/devise.en.yml [hunk 1]
```diff
@@ -1,62 +1,143 @@
...
+      unauthenticated: You need to sign in or sign up before continuing.
+      unconfirmed: You have to confirm your email address before continuing.
     mailer:
       confirmation_instructions:
-        subject: "Confirmation instructions"
-      reset_password_instructions:
-        subject: "Reset password instructions"
-      unlock_instructions:
-        subject: "Unlock instructions"
+        action: Confirm my account
```
Comment (anastasiabalaban) [inline config/locales/devise.en.yml]: Please add  title: "Confirm your Email". It's necessary for my view

File: config/locales/devise.en.yml [hunk 2]
```diff
@@ -1,62 +1,143 @@
...
+        greeting: Hello %{recipient}!
+        message: We're contacting you to notify you that your email is being changed to %{email}.
+        subject: Email Changed
       password_change:
-        subject: "Password Changed"
+        greeting: Hello %{recipient}!
+        message: We're contacting you to notify you that your password has been changed.
+        subject: Password Changed
+      reset_password_instructions:
+        action: Change my password
```
Comment (anastasiabalaban) [inline config/locales/devise.en.yml:63]: title: "Change password"

File: config/locales/devise.uk.yml [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+      invalid: "Не вірний %{authentication_keys} або пароль."
+      last_attempt: "У вас залишилась ще одна спроба перед блокуванням облікового запису."
+      locked: "Ваш обліковий запис заблоковано."
+      not_found_in_database: "Не вірний %{authentication_keys} або пароль."
+      timeout: "Час вашої сесії вичерпано. Будь ласка, увійдіть в систему знову."
+      unauthenticated: "Вам необхідно увійти в систему чи зареєструватися."
+      unconfirmed: "Вам потрібно підтвердити Ваш обліковий запис."
+    mailer:
+      confirmation_instructions:
+        action: "Підтвердити мій аккаунт"
```
Comment (anastasiabalaban) [inline config/locales/devise.uk.yml]: title: "Підтвердження пошти"

File: config/locales/devise.uk.yml [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+      email_changed:
+        greeting: 
+        message: 
+        subject: 
+      password_change:
+        greeting: "Вітаємо, %{recipient}!"
+        message: "Ми намагаємося зв'язатися з Вами, щоб сповістити, що Ваш пароль було змінено."
+        subject: "Пароль змінено"
+      reset_password_instructions:
+        action: "Зміна паролю"
```
Comment (anastasiabalaban) [inline config/locales/devise.uk.yml]: title: "Зміна паролю"
action: "Змінити пароль"

**Review by sashamuliar — APPROVED** [review]

**Review by anastasiabalaban — APPROVED** [review]

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -1,3 +1,14 @@
...
   protect_from_forgery with: :exception
+  before_action :set_locale
+
+  private
+
+  def set_locale
+    I18n.locale = params[:locale] || I18n.default_locale
+  end
+
+  def default_url_options
```
Comment (okohuch) [inline app/controllers/application_controller.rb:11]: Do you need this method? I do not see you use it

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -9,6 +9,8 @@
   </head>
 
   <body>
+  	<%= link_to 'EN', url_for(locale: :en) %>
+ 	<%= link_to 'UA', url_for(locale: :uk) %>
```
Comment (okohuch) [inline app/views/layouts/application.html.erb]: Use tab **2 SPACES**

File: app/controllers/application_controller.rb (hunk 5 above)
Comment (okohuch) [reply app/controllers/application_controller.rb:11]: or is it from gem

File: config/locales/devise.en.yml [hunk 7]
```diff
@@ -1,62 +1,145 @@
...
+        send_me_reset_password_instructions: Send me reset password instructions
+      no_token: You can't access this page without coming from a password reset email. If you do come from a password reset email, please make sure you used the full URL provided.
+      send_instructions: You will receive an email with instructions on how to reset your password in a few minutes.
+      send_paranoid_instructions: If your email address exists in our database, you will receive a password recovery link at your email address in a few minutes.
+      updated: Your password has been changed successfully. You are now signed in.
+      updated_not_active: Your password has been changed successfully.
     registrations:
-      destroyed: "Bye! Your account has been successfully cancelled. We hope to see you again soon."
-      signed_up: "Welcome! You have signed up successfully."
-      signed_up_but_inactive: "You have signed up successfully. However, we could not sign you in because your account is not yet activated."
```
Comment (okohuch) [inline config/locales/devise.en.yml]: why did you move it from `registrations` section?

File: app/controllers/application_controller.rb (hunk 5 above)
Comment (shevchukm) [reply app/controllers/application_controller.rb:11]: yes, I need it
http://guides.rubyonrails.org/i18n.html#setting-the-locale-from-url-params

File: config/routes.rb [hunk 8]
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
-  devise_for :users
+  scope '(:locale)', locale: /en|uk/ do
```
Comment (okohuch) [inline config/routes.rb]: Remove scope with locales from routes.

File: app/controllers/application_controller.rb (hunk 5 above)
Comment (okohuch) [reply app/controllers/application_controller.rb:11]: I see

**Review by viktoriia-bilous — APPROVED** [review]

**Review by okohuch — APPROVED** [review]
